Tests/scripts/infrastructure_tests/common_test.py

```diff
@@ -1,147 +1,124 @@
-    When the pipeline SHAs are in the same order as the commits
-    Then it should return True and the commit that triggered the later pipeline
-
-    """
-    commit1 = mocker.Mock()
-    commit1.id = '1'
-    commit1.created_at = '2020-01-01T00:00:00Z'
-
-    commit2 = mocker.Mock()
-    commit2.id = '2'
-    commit2.created_at = '2020-01-02T00:00:00Z'
-
-    commits = [commit1, commit2]
-    current_sha = '2'
-    previous_sha = '1'
-
-    expected = (True, commit2)
-
-    result = common.are_pipelines_in_order_as_commits(commits, current_sha, previous_sha)
-
-    assert result == expected
-
-
-def test_are_pipelines_in_order_as_commits_false(mocker):
-    """
-    Given a list of commits and two pipeline SHAs
-    When the pipeline SHAs are not in the same order as the commits
-    Then it should return False and None
-    """
-    commit1 = mocker.Mock()
-    commit1.id = '1'
-    commit1.created_at = '2020-01-01T00:00:00Z'
-
-    commit2 = mocker.Mock()
-    commit2.id = '2'
-    commit2.created_at = '2020-01-02T00:00:00Z'
-
-    commits = [commit1, commit2]
-    current_sha = '1'
-    previous_sha = '2'
-
-    expected = (False, None)
-
-    result = common.are_pipelines_in_order_as_commits(commits, current_sha, previous_sha)
-    # there is a problem to test if side effect that is None is equal to None, so for now its removed from the assert
-    assert result[0] == expected[0]
-
-
-def test_is_pivot_first_pipeline(mocker):
+    Given:
+        - Two pipelines with their created_at dates.
+    When:
+        - are_pipelines_in_order is called on them.
+    Then:
+        - It should return the expected result.
+          scenario 1: pipeline1.created_at > pipeline2.created_at -> False
+          scenario 2: pipeline1.created_at < pipeline2.created_at -> True
     """
-    Given a pipeline id, list of pipelines and commits
-    When the pipeline is the first in the list
-    Then it should return None, None
-    """
-    pipeline_id = '1'
-    pipelines = [mocker.Mock(id=1)]
-    commits = [mocker.Mock()]
+    pipeline1 = mocker.Mock()
+    pipeline1.id = '1'
+    pipeline1.created_at = pipeline1_date
 
-    expected = (None, None)
+    pipeline2 = mocker.Mock()
+    pipeline2.id = '2'
+    pipeline2.created_at = pipeline2_date
 
-    result = common.is_pivot(pipeline_id, pipelines, commits)
+    result = are_pipelines_in_order(pipeline1, pipeline2)
 
     assert result == expected
 
 
-def test_is_pivot_pipeline_not_in_list(mocker):
+@pytest.mark.parametrize('pipeline1_status, pipeline2_status, expected', [
+    ('success', 'failed', True),
+    ('failed', 'success', False),
+    ('success', 'success', None),
+    ('failed', 'in progress', None)
+])
+def test_is_pivot(mocker, pipeline1_status, pipeline2_status, expected):
     """
-    Given a pipeline id, list of pipelines and commits
-    When the pipeline id is not in the list of pipelines
-    Then it should return None, None
+    Given:
```

Comment:
highly complex test, consider splitting to separate tests.
The scenario are not detailed enough.